### DISCUSSION TITLE
Are the '@babel/preset-typescript' options in the provided babel presets sensible?

### DISCUSSION THREAD

**Discussion by jennysharps** [shakacode/shakapacker#246] [General]
At the moment, the typescript babel preset looks like this:

https://github.com/shakacode/shakapacker/blob/master/package/babel/preset.js#L33
```javascript
[
  '@babel/preset-typescript',
  { allExtensions: true, isTSX: true }
]
```

According to this issue https://github.com/babel/babel/issues/8924, `isTSX` & `allExtensions` being set to `true` means the entire codebase is expected to be typescript.  I'm trying to use the provided preset, but as my codebase is a combination of `js` & `ts`, I get errors.

Would it not be more sensible of a default to pass in no options in the preset?  Especially given the fact that one of the benefits of using typescript is the ability to introduce it incrementally to existing codebases.

**Comment by justin808** [comment]
I agree! This change was introduced here, many years ago.
https://github.com/shakacode/shakapacker/blame/master/package/babel/preset.js#L33

This would be a "breaking" change, so it's great for the upcoming v7!

**Reply by justin808** [reply]
#267 is the tracking issue. Thanks, @jennysharps!

**Comment by ahangarha** [comment]
@jennysharps 
Please check #273
